src/stack.rs

```diff
@@ -0,0 +1,285 @@
+    /// let mut stack = Stack::new();
+    /// stack.push(1);
+    /// assert!(!stack.is_empty());
+    /// ```
+    pub fn push(&mut self, elem: T) {
+        if self.is_empty() {
+            self.first = Some(Box::new(Node::new(elem)));
+        } else {
+            let mut new_node = Box::new(Node::new(elem));
+            new_node.next = self.first.take();
+            self.first = Some(new_node);
+        }
+    }
+
+    /// Returns the size (depth) of the stack. It iterates over all elements to do so and is therefore not super fast.
+    /// ```rust
+    /// use data_structure_with_colin::stack::Stack;
+    /// let mut stack = Stack::new();
+    /// stack.push(1);
+    /// stack.push(2);
+    /// assert_eq!(stack.size(), 2);
+    /// ```
+    pub fn size(&self) -> i32 {
+        if self.is_empty() {
+            return 0;
+        } else {
+            return self.first.as_ref().unwrap().size();
+        }
+    }
+
+    /// Checks if a stack contains a specific element. It iterates over the elements until it finds the searched one
+    /// or the end is reached.
+    /// ```rust
+    /// use data_structure_with_colin::stack::Stack;
+    /// let mut stack = Stack::new();
+    /// stack.push(1);
+    /// assert!(!stack.contains(2));
+    /// assert!(stack.contains(1));
+    /// ```
+    pub fn contains(&self, value: T) -> bool {
+        return if self.is_empty() {
+            false
+        } else if self.first.as_ref().unwrap().value == value {
+            true
+        } else {
+            self.first.as_ref().unwrap().contains(value)
+        };
+    }
+
+    /// Pops the first element of a stack. Remember that the stacks first element is always the last one that got inserted (LIFO-Principle).
+    /// ```rust
+    /// use data_structure_with_colin::stack::Stack;
+    /// let mut stack = Stack::new();
+    /// stack.push(2);
+    /// stack.push(1);
+    /// assert_eq!(stack.pop(), Some(1));
+    /// assert!(!stack.is_empty());
+    /// ```
+    pub fn pop(&mut self) -> Option<T> {
+        if self.is_empty() {
+            return None;
+        } else {
+            let mut old_head = self.first.take();
+            self.first = old_head.as_mut().unwrap().next.take();
+            Some(old_head.unwrap().value)
+        }
+    }
+
+    /// Checks if the current stack is empty or not.
+    /// ```rust
+    /// use data_structure_with_colin::stack::Stack;
+    /// let mut stack = Stack::new();
+    /// assert!(stack.is_empty());
+    /// stack.push(1);
+    /// assert!(!stack.is_empty());
+    /// ```
+    pub fn is_empty(&self) -> bool {
+        self.first.is_none()
+    }
+
+    /// Returns a non-owning iterator for the stack which iterates over the element in the LIFO way.
+    pub fn iter(&self) -> Iter<T> {
+        Iter { head: &self.first }
+    }
+}
+
+impl<T: Eq> Node<T> {
+    fn new(value: T) -> Self {
+        Node { next: None, value }
+    }
+
+    fn size(&self) -> i32 {
```

Comment:
I'm not happy with a non storing size function here tbh, could you store the size in a variable pls

Reply:
Imagine a stack with 10000+ elements, the size-method does not scale and all just for adding numbers. Do you know what i mean?